codice/ddf
# DDF-2051 Allows system to be configured and run without a users.properties file.

#### What does this PR do?
* Fixes `SolrClientFactory` so that it will retry connection to cores.
* Prevents failed validation of `users.properties` if not present.
* Fixes LDAP claims handlers to work with DN of provided X500 Principals so site certificate-based authN/Z will correctly find NPE entries not co-located with user entries in LDAP systems.

This will allow a system to be configured such that the `users.properties` file can be removed.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@tbatie 
@ryeats 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@stustison

#### How should this be tested?
* Full build with unit/itests
* Install and run with a non-localhost site certificate
* Add embedded LDAP or configure an external LDAP
* Add a user to LDAP with either
    * a DN that matches the DN of the site certificate if it is fully qualified
    * or a CN that matches the CN of the site certificate, placed in the user root DN, if it does not have a FQDN
* Update the Web Context Policy Manager, changing the root Context Realm from `karaf` to `ldap` 
* Remove `users.properties`
* Restart system and verify operation of admin (make sure to test configuration export functionality) and search features

#### Any background context you want to provide?
N/A

#### What are the relevant tickets?
DDF-2051

#### Screenshots (if appropriate)
N/A

#### Checklist:
